### PR TITLE
bnxt_re/lib: Code refactoring in the resource allocation verbs

### PR DESCRIPTION
**Pull request by selvintxavier**
Implements few helper functions for better code maintainability. Allocates the queue memory and the shadow queue memory together and manage the  pointer for accessing the queue memory and shadow queue memory. Also splits some big functions to smaller sub functions.

### CODE REVIEW

File: providers/bnxt_re/main.h [hunk 1]
```diff
@@ -614,4 +620,14 @@ static inline void bnxt_re_sub_sec_busy_wait(uint32_t nsec)
 }
 
 #define BNXT_RE_HW_RETX(a) ((a)->comp_mask & BNXT_RE_COMP_MASK_UCNTX_HW_RETX_ENABLED)
+
+static inline uint32_t llog2(uint32_t x)
```
Comment (rleon) [inline providers/bnxt_re/main.h:626]: we already have this function, "git grep ilog32"

Comment (selvintxavier) [reply providers/bnxt_re/main.h:626]: > we already have this function, "git grep ilog32"

Thanks. I missed searching for log32. Will update the PR.

File: providers/bnxt_re/main.h [hunk 2]
```diff
@@ -554,6 +557,11 @@ static inline void bnxt_re_jqq_mod_last(struct bnxt_re_joint_queue *jqq,
 	jqq->last_idx = jqq->swque[idx].next_idx;
 }
 
+static inline uint32_t bnxt_re_get_diff(uint64_t cmask)
```
Comment (rleon) [inline providers/bnxt_re/main.h:560]: 1. It is not used
2. There is no value in function which simply returns defind

Comment (selvintxavier) [reply providers/bnxt_re/main.h:560]: Agree that it is not useful with the current patch. I will remove it.
I have a follow up series which avoid this diff for newer adapters, that includes kernel driver changes. We will have to check the compatibility mask for adding the diff. I will re-introduce the function for that series.

**Pull request merged**